### ISSUE TITLE
Manual Implementation of MLP for MNIST Classification

### ISSUE THREAD

**Issue by aidyn-kittrell** [aidyn-kittrell/Deep-Digits-Duel-MLP-vs-CNN#1]
**Goal**: Manually implement the following components (functions):
1. Forward propagation function
2. Backward propagation function (including gradient calculations)
3. Train function (for parameter updates of MLP neural network by loss)
4. Sigmoid activation function (for the 1st layer)
5. Softmax activation function (for the 2nd layer)
6. Cross-entropy function (for calculating loss)
7. Main function (for training and testing designed MLP by train dataloader and test dataloader respectively)

**Requirements**:
* Do NOT user any deep learning APIs (e.g., torch.nn, tf.keras, or other pre-built neural network layers)
* Model Architecture, a 2-layer neural network with the following structure:
    1. 1st fully connected layer (student define the size) + Sigmoid activation
    2. 2nd fully connected layer + 10 neurons with Softmax activation (for 10 classification)
* Implementation Constraints:
    1. Forward Propagation:
        * Input: Batch of flattened MNIST images(shape[batch_size, 784])(input size 28x28=784)
        * Output: Prediction probabilities after passing through the two layers
    2. Backward Propagation:
        * Input: Batch of images, ground-truth labels, and prediction probabilities from forward propagation
        * In this pass, compute gradients averaged over the current batch (e.g., divide gradients by the batch size)
        * Update parameters (W1, b1, W2, b2) via gradient descent
        * Use the chain rule to manually derive gradients for weights/biases for both layers
    3. Train Function:
        * Use batch-based training (i.e., process data in batches of size 128 as provided by the dataloader
        * Input: Training data (images and labels)
        * Iterate over batches of data (maximum 469 iterations per epoch, maximum 100 epochs)
        * Call forward() and backward() in sequence
        * Track and return the training loss
* Additional Specifications:
    1. Use random initialization for weights/biases (e.g., Gaussian distribution for weights, zeros for biases)
    2. Report/Print test accuracy after training (testing code will be provided)
    3. The code should be able to print out the training loop indicator with loss/accuracy logging
    4. Use Python as the programming language
